r/R/dataset-format.R

```diff
@@ -122,6 +122,30 @@ CsvFileFormat$create <- function(...,
                                  opts = csv_file_format_parse_options(...),
                                  convert_options = csv_file_format_convert_opts(...),
                                  read_options = csv_file_format_read_opts(...)) {
+
+  # Evaluate opts first to catch any unsupported arguments
+  force(opts)
+
+  options <- list(...)
+  schema <- options[["schema"]]
+
+  column_names <- read_options$column_names
+  schema_names <- names(schema)
+
+  if (!is.null(schema) & !identical(schema_names, column_names)) {
+
+    # Element wise comparison and set differnce of column_names and names in schema
+    mismatch_colnames <- column_names[match(column_names, schema_names, nomatch = 0) != seq(column_names)]
+    not_in_schema <- setdiff(schema_names, column_names)
+
+    abort(
+      paste(
+        "column_names must match schema-names:",
+        deparse1(c(mismatch_colnames, not_in_schema))
+      )
```

Comment:
```suggestion
      paste(
        "Values in `column_names` must match schema field names"
      )
```

Reply:
I've just removed the pasting of the fields as it's a little unclear which are missing from which, so best to leave out to reduce ambiguity!